### PR TITLE
convai2 updates

### PR DESCRIPTION
**Pull request by alexholdenmiller**
- add code to reproduce best seq2seq model
- opensubtitles doubling fixed
- seq2seq change some defaults (esp--switch to sgd), prevent ranker from duplicating params saved and allowing ranking after training without, add support for non-length-sorted inputs

### CODE REVIEW

**Comment by alexholdenmiller** [comment]
adding this is first step towards convai2 folder, will need to add documentation next

**Comment by alexholdenmiller** [comment]
sure thing!

File: projects/convai2/README.md [hunk 1]
```diff
@@ -20,14 +20,7 @@ For example, you can download and evaluate a pre-trained seq2seq model (based on
...
 
-You can submit your model by creating a PR with a folder for your model in the `submissions` folder in this directory.
-You do not have to submit your competition model before the evaluation deadlines, but feel free to submit additional simple baselines for others to compare to or use to get started.
-While you don't *have* to train your model in ParlAI, we encourage you to provide full model code and a similar `train.sh` file so that others can reproduce your results.
-
-In order to have your trained model evaluated for the competition, however, you will have to submit code similar to the `eval.sh` script in the `baselines/seq2seq` folder.
-We will use the eval_model command in the same way, using the `valid` datatype for the validation leaderboard and a hidden test set for final evaluation (you don't need to support batching or CUDA with your model, though).
-This will require at least writing an inference-time wrapper according to the ParlAI format.
-If you need help hosting your trained model file, contact @alexholdenmiller (email `ahm at fb`).
+Submission guidelines will be released further into the competition.
```
Comment (jaseweston) [inline projects/convai2/README.md]: can we just remove this section completely?

File: projects/convai2/README.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+
+## Examples
+
+You can run examples of training on this task in the `baselines` folder in this directory.
+
+For example, you can download and evaluate a pre-trained seq2seq model (based on ParlAI's implementation in parlai/agents/seq2seq) using `baselines/seq2seq/eval.sh`, or train your own with the same parameters using the `train.sh` file in that directory.
+
+
+## Contributing for the final evaluation
```
Comment (jaseweston) [inline projects/convai2/README.md]: can we just remove this?

File: projects/convai2/README.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
+# ConvAI2 Competition for NIPS 2018
+
+This directory contains code and examples for the ConvAI2 competition.
+See the [website](http://convai.io/) for more details.
+
+The ConvAI2 dataset is already in ParlAI in parlai/agents/tasks/convai2 and is accessible by using the command line parameter `--task convai2`.
+
+To see this data, try:
```
Comment (jaseweston) [inline projects/convai2/README.md]: E.g. to see the training data, try:

File: projects/convai2/README.md [hunk 4]
```diff
@@ -0,0 +1,35 @@
+# ConvAI2 Competition for NIPS 2018
+
+This directory contains code and examples for the ConvAI2 competition.
+See the [website](http://convai.io/) for more details.
+
+The ConvAI2 dataset is already in ParlAI in parlai/agents/tasks/convai2 and is accessible by using the command line parameter `--task convai2`.
```
Comment (jaseweston) [inline projects/convai2/README.md]: remove "already"

File: parlai/agents/seq2seq/seq2seq.py [hunk 5]
```diff
@@ -538,7 +547,7 @@ def batch_act(self, observations):
         if is_training:
             report_freq = 0
         else:
-            report_freq = 0.01
+            report_freq = 0.001
```
Comment (jaseweston) [inline parlai/agents/seq2seq/seq2seq.py]: it seems like this should be a param, you keep changing it?

Comment (alexholdenmiller) [reply parlai/agents/seq2seq/seq2seq.py]: I mean it's just a logging param, so changing it via the cmdline is kind of annoying anyways

**Comment by alexholdenmiller** [comment]
lmk what you think about train.py and eval.py instead of the .sh versions

File: parlai/agents/seq2seq/seq2seq.py (hunk 5 above)
Comment (jaseweston) [reply parlai/agents/seq2seq/seq2seq.py]: you changing it every PR is annoying...!! lol

Comment (jaseweston) [reply parlai/agents/seq2seq/seq2seq.py]: you would have a default on the param, so it wouldnt change anything

**Comment by alexholdenmiller** [comment]
going to land because something in it is blocking emily but feel free to comment more changes

**Pull request merged**